apps/io_tester/io_tester.cc

```diff
@@ -207,8 +209,12 @@ struct job_config {
     ::options options;
     // size of each individual file. Every class and every shard have its file, so in a normal
     // system with many shards we'll naturally have many files and that will push the data out
-    // of the disk's cache
+    // of the disk's cache. An exception to that rule is unlink_class_data, that creates files_count
+    // files with file_size/files_count.
```

Comment:
Why this exception?

Reply:
In `struct convert<job_config>::decode()` we have the following lines:

```
// The data_size is used to divide the available (and effectively
// constant) disk space between workloads. Each shard inside the
// workload thus uses its portion of the assigned space.
if (node["data_size"]) {
    cl.file_size = node["data_size"].as<byte_size>().size / smp::count;
} else {
    cl.file_size = 1ull << 30; // 1G by default
}
```

If `unlink_class_data` created `files_count` files with `file_size` then the assumption about `data_size` config field would be wrong.

As far as I understand `data_size`, it divides disk space between workloads. Each shard receives only a piece of that `data_size`. To ensure, that `unlink_class_data` uses only that piece, the calculation of a single file in that job is performed as follows:
`const auto fsize = _config.file_size / files_count();`

That's the reason why the comment in `struct job_config` was adjusted to:

```
// size of each individual file. Every class and every shard have its file, so in a normal
// system with many shards we'll naturally have many files and that will push the data out
// of the disk's cache. An exception to that rule is unlink_class_data, that creates files_count
// files with file_size/files_count.
uint64_t file_size;
```

Is it wrong? Which part should be corrected? Should the comment explain the origin of that exception or should this exception be removed altogether? Or maybe the calculation of `file_size` in `struct convert<job_config>::decode()` should be adjusted?

Reply:
Ah, you're right. I forgot that `file_size` here is actually the `data_size` from yaml :( Let's keep this assumption for unlink class too